org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/AbstractServiceReferenceAssert.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.assertj.servicereference;
+
+import static org.osgi.test.assertj.bundle.BundleAssert.BUNDLE;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.assertj.core.api.AbstractComparableAssert;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.assertj.core.api.InstanceOfAssertFactory;
+import org.assertj.core.api.ListAssert;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.osgi.test.assertj.bundle.BundleAssert;
+import org.osgi.test.assertj.dictionary.DictionaryAssert;
+
+public abstract class AbstractServiceReferenceAssert<SELF extends AbstractServiceReferenceAssert<SELF, ACTUAL>, ACTUAL extends ServiceReference<?>>
+	extends AbstractComparableAssert<SELF, ACTUAL> {
+
+	protected AbstractServiceReferenceAssert(ACTUAL actual, Class<?> selfType) {
+		super(actual, selfType);
+	}
+
+	public DictionaryAssert<String, Object> hasServicePropertiesThat() {
+		return new DictionaryAssert<String, Object>(actual.getProperties());
+	}
```

Comment:
This depends on us moving the baseline OSGi support to R7 and we're not 100% sure if we're going down that yet - I think it would be courtesy to wait to hear from @rotty3000 for his opinion as I know he's using osgi-test and he may require R6 compatibility.

Leave it this way for now, but make sure you don't throw away the `ServiceReferenceAsDictionary` implementation in case we need to roll back to R6.

Reply:
https://gist.github.com/stbischof/943a8ed64470a385d8224313289568ad

Reply:
For now I would rather keep it R6 compatible.

Reply:
OK, but then we should not make API which we won't need in the future when we move to R7. So we need a method to make a Dictionary from a ServiceReference. Then one can assert on that Dictionary. When we move to R7, the implementation of the method to make a Dictionary from a ServiceReference changes.

Reply:
Agreed. @stbischof - looks like you'll need to dust off that `ServiceReferenceAsDictionary` implementation and slot it in here for now. :smile:

Reply:
i added 
```
public static <K, V> Dictionary<String, Object> asDictionary(ServiceReference<?> serviceReference) {
		return new ServiceReferenceAsDictionary(serviceReference);
		
```